src/Multi.php

```diff
@@ -58,6 +59,30 @@ public static function zipLongest(iterable ...$iterables): \Generator
         }
     }
 
+    /**
+     * Iterate multiple iterable collections simultaneously.
+     *
+     * Make an iterator that aggregates items from multiple iterators.
+     * Similar to Python's zip_longest function
+     *
+     * Iteration continues until the longest iterable is exhausted.
+     * For uneven lengths, the exhausted iterables will produce $filler value for the remaining iterations.
+     *
+     * @param mixed $filler
+     * @param iterable<mixed> ...$iterables
+     *
+     * @return \Generator<array<mixed>>
+     */
+    public static function zipFilled($filler, iterable ...$iterables): \Generator
+    {
+        $iterator = new JustifyMultipleIterator(...$iterables);
+        $iterator->setFiller($filler);
```

Comment:
Suggestion: Perhaps consider a new "Filled" iterator or something that would take the filler as a constructor parameter along with the iterables. Using a setter mutates the iterator. It is generally a good idea to consider objects as immutable whenever possible. While an iterator technically does have mutable state with its internal pointer, we can avoid additional unnecessary mutations by requiring all parameters at the time of construction.

Reply:
Done.